### PR TITLE
Fixes for issue 12982 Android cross-build fails when compiling cee_wks

### PR DESCRIPTION
**Pull request by cydhaselton**
The cross build script needs to be updated with the correct supporting package versions or download and build will fail.

Copying @qmfrederik and @jkotas  @sdmaclea

### CODE REVIEW

File: src/inc/volatile.h [hunk 1]
```diff
@@ -120,7 +120,7 @@ T VolatileLoad(T const * pt)
     STATIC_CONTRACT_SUPPORTS_DAC_HOST_ONLY;
 
 #ifndef DACCESS_COMPILE
-#if defined(_ARM64_) && defined(__GNUC__)
+#if defined(_ARM64_) && defined(__GNUC__) && !(defined(__ANDROID__))
```
Comment (jkotas) [inline src/inc/volatile.h]: Would it be better to check against GCC version that added support for `__atomic_load`?

Comment (cydhaselton) [inline src/inc/volatile.h]: Not sure; plus I think the Android NDK/toolchains are moving to Clang and away from GCC.

Comment (sdmaclea) [inline src/inc/volatile.h]: It seemed strange to check GCC since we generally use Clang.  I kept it to follow the existing convention of the file

Comment (cydhaselton) [inline src/inc/volatile.h]: Okay to keep the `__ANDROID__` check then?  Let me know; I can search for an appropriate Clang check if any.

Comment (jkotas) [inline src/inc/volatile.h]: It would be nice to make this compiler version check. Which compiler and version are you using for Android right now?

Comment (cydhaselton) [inline src/inc/volatile.h]: clang 3.8

Comment (jkotas) [inline src/inc/volatile.h]: Would `defined(__clang__) && __has_builtin(__atomic_load)` work? We use `__has_builtin` in other places.

Comment (cydhaselton) [inline src/inc/volatile.h]: Checking now…

Comment (cydhaselton) [inline src/inc/volatile.h]: @jkotas Worked. Changed and committed.

**Comment by cydhaselton** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test

**Comment by cydhaselton** [comment]
@janvorli Fixed.

File: cross/android/arm64/toolchain.cmake [hunk 2]
```diff
@@ -21,6 +21,7 @@ find_program(CMAKE_OBJDUMP ${TOOLCHAIN_PREFIX}objdump)
 
 add_compile_options(--sysroot=${CROSS_ROOTFS})
 add_compile_options(-fPIE)
+add_compile_options(-D__ANDROID__)
```
Comment (jkotas) [inline cross/android/arm64/toolchain.cmake]: This is not necessary anymore

File: src/inc/volatile.h [hunk 3]
```diff
@@ -120,7 +120,7 @@ T VolatileLoad(T const * pt)
     STATIC_CONTRACT_SUPPORTS_DAC_HOST_ONLY;
 
 #ifndef DACCESS_COMPILE
-#if defined(_ARM64_) && defined(__GNUC__)
+#if defined(_ARM64_) && defined(__GNUC__) && defined(__clang__) && __has_builtin(__atomic_load)
```
Comment (jkotas) [inline src/inc/volatile.h]: `defined(__GNUC__)` is not necessary - it is always defined for clang.

**Comment by cydhaselton** [comment]
@jkotas Fixed

**Comment by jkotas** [comment]
Windows builds are failing with: `volatile.h(123): warning C4067: unexpected tokens following preprocessor directive - expected a newline`

I guess `__has_builtin` has to be in clang specific block - something like:

```
#if defined(_ARM64_) && defined(__clang__)
...
#if __has_builtin(__atomic_load)
      if((1 << sizeof(T)) & lockFreeAtomicSizeMask) 
     {
         __atomic_load((T volatile const *)pt, &val, __ATOMIC_ACQUIRE);
     }
     else
#endif
...
```

**Comment by cydhaselton** [comment]
Thanks for digging into that @jkotas 
Changes made and committed

**Comment by cydhaselton** [comment]
@jkotas Android cross-build errors with (__clang__) and __has_builtin(__atomic_load) conditionals, apologies I thought I tested this. 
I'm back to (__ANDROID__) or I can use (!defined(__BIONIC__) if that would be more appropriate.

File: src/inc/volatile.h [hunk 4]
```diff
@@ -120,13 +120,15 @@ T VolatileLoad(T const * pt)
...
+#if defined(_ARM64_) && defined(__clang__)
     T val;
     static const unsigned lockFreeAtomicSizeMask = (1 << 1) | (1 << 2) | (1 << 4) | (1 << 8);
+#if __has_builtin(__atomic_load)
     if((1 << sizeof(T)) & lockFreeAtomicSizeMask)
     {
         __atomic_load((T volatile const *)pt, &val, __ATOMIC_ACQUIRE);
     }
+#endif
     else
```
Comment (sdmaclea) [inline src/inc/volatile.h]: This stray else looks wrong.

**Comment by cydhaselton** [comment]
@jkotas Can you revert?  See my earlier comments about the conditionals...

**Pull request merged**